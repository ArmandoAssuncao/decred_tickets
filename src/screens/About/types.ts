import { IScreenProps } from 'containers/Root/types'

export interface IConnectedProps {
}

export interface IActionCreators {
}

export interface IProps extends IScreenProps, IConnectedProps, IActionCreators {
}

export interface IState {
}
