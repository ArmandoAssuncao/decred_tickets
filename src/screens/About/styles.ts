import { ViewStyle, TextStyle } from 'react-native'
import EStyleSheet from 'react-native-extended-stylesheet'

export default EStyleSheet.create({
    container: {
        backgroundColor: '$primaryColor',
        flex: 1,
    } as ViewStyle,
    scrollContentContainer: {
        alignItems: 'center',
        flex: 1,
        padding: '0.4rem',
    } as ViewStyle,
    text: {
        color: '$textColorSecond',
    } as TextStyle,
    sourceContainer: {
        flexDirection: 'row',
    } as ViewStyle,
    textContainer: {
        margin: '0.9rem',
    } as ViewStyle,
    textLink: {
        color: '$textColorLink',
        marginLeft: '0.5rem',
        textDecorationLine: 'underline',
    } as TextStyle,
    walletContainer: {
        alignItems: 'center',
    } as TextStyle,
})
