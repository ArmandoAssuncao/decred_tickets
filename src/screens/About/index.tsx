import React from 'react'
import { Clipboard, Linking, ScrollView, TouchableOpacity, View } from 'react-native'
import { NavigationScreenConfig } from 'react-navigation'
import { connect, MapDispatchToProps } from 'react-redux'
import { bindActionCreators } from 'redux'
import { IAppState } from 'store/types'

import i18n from 'config/i18n'
import constants from 'config/constants'

import { IActionCreators, IConnectedProps, IProps, IState, } from './types'
import styles from './styles'

import { Text } from 'components'

const mapStateToProps = (_state: IAppState): IConnectedProps =>
  ({})

const mapDispatchToProps = (dispatch: any): MapDispatchToProps<IActionCreators, IProps> =>
    bindActionCreators(
        {},
        dispatch
    )

export class About extends React.PureComponent<IProps, IState> {
    static navigationOptions: NavigationScreenConfig<any> = {
        title: i18n.t('screens.about.title'),
    }

    private onPressOpenLink = (): void => {
        const githubLink = constants.REPO_URL
        Linking.openURL(githubLink)
        .catch(() => alert(i18n.t('app.errorOpenURL', {link: githubLink})))
    }

    private onPressCopyToClipboard = (): void => {
        Clipboard.setString(constants.DECRED_WALLET_ADDRESS)
        alert(i18n.t('app.clipboardCopy'))
    }

    render(): JSX.Element {
        return (
            <ScrollView style={styles.container} contentContainerStyle={styles.scrollContentContainer}>
                <View style={styles.textContainer}>
                    <Text m="Decred Tickets" size="xl" style={styles.text}>{constants.APP_NAME}</Text>
                </View>
                <View style={styles.textContainer}>
                    <Text id={'screens.about.createdBy'} m="Created By" style={styles.text} />
                </View>
                <View style={styles.textContainer}>
                    <View style={styles.sourceContainer}>
                        <Text id={'screens.about.source'} m="Source Code" style={styles.text} />
                        <TouchableOpacity onPress={this.onPressOpenLink}>
                            <Text id={'app.github'} m="Github" style={styles.textLink} />
                        </TouchableOpacity>
                    </View>
                </View>
                <View style={styles.textContainer}>
                    <TouchableOpacity onPress={this.onPressCopyToClipboard} style={styles.walletContainer}>
                        <Text id={'screens.about.leave'} m="If you liked it, leave DCRs ;-)" style={styles.text} />
                        <Text m="" type="bold" style={styles.text}>{constants.DECRED_WALLET_ADDRESS}</Text>
                        <Text id={'app.clickToCopy'} m="Click To Copy" size="xs" style={styles.text} />
                    </TouchableOpacity>
                </View>
            </ScrollView>
        )
    }
}

export default connect(mapStateToProps, mapDispatchToProps)(About)
