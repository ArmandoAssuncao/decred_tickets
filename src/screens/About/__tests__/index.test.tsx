import * as React from 'react'
import { shallowStore } from 'test/enzyme'

import About from '../index'

describe('About test render', () => {
    const store = {}

    const wrapper = shallowStore(<About />, store)

    it('Testing snapshot', () => {
        expect(wrapper).toMatchSnapshot()
    })

    it('Testing instance of component', () => {
        expect(wrapper.instance()).toBeInstanceOf(About.WrappedComponent)
    })

    it('has 1 ScrollView', () => {
        expect(wrapper.find('ScrollView')).toHaveLength(1)
    })

    it('has 5 Component', () => {
        expect(wrapper.find('Component')).toHaveLength(5)
    })

    it('has 7 Text', () => {
        expect(wrapper.find('Text')).toHaveLength(7)
    })

    it('has 2 TouchableOpacity', () => {
        expect(wrapper.find('TouchableOpacity')).toHaveLength(2)
    })
})
