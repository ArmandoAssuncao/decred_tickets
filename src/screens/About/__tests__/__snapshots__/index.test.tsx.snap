// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`About test render Testing snapshot 1`] = `
<ScrollView
  contentContainerStyle={
    Object {
      "alignItems": "center",
      "flex": 1,
      "padding": 6.4,
    }
  }
  style={
    Object {
      "backgroundColor": "#0C1E3E",
      "flex": 1,
    }
  }
>
  <Component
    style={
      Object {
        "margin": 14.4,
      }
    }
  >
    <Text
      m="Decred Tickets"
      size="xl"
      style={
        Object {
          "color": "#FFF",
        }
      }
      type="normal"
    >
      Decred Tickets
    </Text>
  </Component>
  <Component
    style={
      Object {
        "margin": 14.4,
      }
    }
  >
    <Text
      id="screens.about.createdBy"
      m="Created By"
      size="md"
      style={
        Object {
          "color": "#FFF",
        }
      }
      type="normal"
    />
  </Component>
  <Component
    style={
      Object {
        "margin": 14.4,
      }
    }
  >
    <Component
      style={
        Object {
          "flexDirection": "row",
        }
      }
    >
      <Text
        id="screens.about.source"
        m="Source Code"
        size="md"
        style={
          Object {
            "color": "#FFF",
          }
        }
        type="normal"
      />
      <TouchableOpacity
        activeOpacity={0.2}
        onPress={[Function]}
      >
        <Text
          id="app.github"
          m="Github"
          size="md"
          style={
            Object {
              "color": "#6495ED",
              "marginLeft": 8,
              "textDecorationLine": "underline",
            }
          }
          type="normal"
        />
      </TouchableOpacity>
    </Component>
  </Component>
  <Component
    style={
      Object {
        "margin": 14.4,
      }
    }
  >
    <TouchableOpacity
      activeOpacity={0.2}
      onPress={[Function]}
      style={
        Object {
          "alignItems": "center",
        }
      }
    >
      <Text
        id="screens.about.leave"
        m="If you liked it, leave DCRs ;-)"
        size="md"
        style={
          Object {
            "color": "#FFF",
          }
        }
        type="normal"
      />
      <Text
        m=""
        size="md"
        style={
          Object {
            "color": "#FFF",
          }
        }
        type="bold"
      >
        DsTMYzhVtsY62K7EUUv32VbrBQpRJCdbarB
      </Text>
      <Text
        id="app.clickToCopy"
        m="Click To Copy"
        size="xs"
        style={
          Object {
            "color": "#FFF",
          }
        }
        type="normal"
      />
    </TouchableOpacity>
  </Component>
</ScrollView>
`;
